### ISSUE TITLE
Problems in the transition with a large number of files

### ISSUE THREAD

**Issue by KOT040188** [luspi/photoqt-dev#43]
Sorry for bad English, I use google translate. I very like PhotoQt. There are problems with viewing large number of images in one folder. I installed kubuntu 14.04 and icons ursa-icons (ppa: noobslab / icons). I moved to the folder / usr / share / icons / Ursa-icons / 128x128 / apps /, where 2224 file and start of hits. After a few seconds PhotoQt fell. I started it again, this time it did not fall, but had terrible problems with the transition from one file to another. Pressing the left and right arrows PhotoQt passed only ~ 5 images, the rest ignored. I click the mouse on the thumbnails on the other images were similar problems (transition was at 2, 3, 4 images).

**Comment by luspi** [comment]
Hm, I downloaded the Ursa icon theme, but I wasn't able to reproduce your issue, it works fine for me...

Could you start photoqt from a terminal with the '-v' switch (i.e: 'photoqt -v') and then post back here the output you get?

**Comment by KOT040188** [comment]
If you open a photoqt abiword.pn, then pressing the right arrow nothing happens. Clicking on the arrow to the left photoqt jumped to the other 6 icons, starting with accessories-text-editor.png, and fixated on them ...
kot@kot-G31M-ES2L:~$ photoqt -v

PhotoQt v1.2 -  Copyright (C) 2013, Lukas Spies (Lukas@photoqt.org), License: GPLv2 (or later)
PhotoQt is a fast, simple, good looking, yet powerfull and highly configurable image viewer.

Usage: photoqt [options|file]

Options:
        --h, --help             This help message

>> Start-up options:

        --start-in-tray         Start PhotoQt hidden to the system tray
        --no-thumbs             Don't load thumbnails (Navigation through folder is still possible)

>> Remote Controlling:

        --open                  Opens the open file dialog (also shows PhotoQt if hidden)
        --toggle                Toggles PhotoQt - hides PhotoQt if visible, shows if hidden
        --show                  Shows PhotoQt (does nothing if already shown)
        --hide                  Hides PhotoQt (does nothing if already hidden)

>> Remote controlling w/ filename needed:

        --no-thumbs [filename]  Don't load thumbnails (Navigation through folder is still possible)
        --thumbs [filename]     Reversing a '--no-thumbs' (thumbnails are enabled by default)

>> Debuging:

        --v, --verbose          Enabling debug messages


   Enjoy PhotoQt :-)

**Comment by KOT040188** [comment]
http://pastebin.com/YKfEG6jw Do the same as the option in the previous comments.

**Comment by luspi** [comment]
Yes, sorry about that, I indeed meant 'photoqt --v'.

Also, I found the problem: PhotoQt wasn't able to properly handle symlinks (symlinks don't actually contain any information, but only point to some other file), and ~2000 of the files in the icons directory are symlinks. As PhotoQt checked for the position of the symlink in the directory it received the position of the file pointed to. This led to the seemingly random jumps back and forth in the directory.

The fix was, to simply extract a list of filenames explicitely when loading a directory and then to use this list to get the position of the file in the directory: 02c95089d412ecb5d876100c3d7be8b056bbe631

Thanks for helping me to track down the problem (actually I even found another one along the way)!

As I commented on the other issue, the only way for you to immediately get the updated version, would be, to compile it yourself from the code here on GitHub. Otherwise, unfortunately, you'd need to wait for v1.2.1 and the update in the ubuntu repository.
